actions/xmpp

```diff
@@ -230,6 +234,67 @@ def subcommand_change_domainname(arguments):
     action_utils.service_start('ejabberd')
 
 
+def subcommand_add_letsencrypt(arguments):
+    """Start using letsencrypt certificate if it matches configured domain."""
+    with open(EJABBERD_CONFIG, 'r') as file_handle:
+        conf = ruamel.yaml.round_trip_load(file_handle, preserve_quotes=True)
+
+    if arguments.domain in conf['hosts']:
+        cert_folder = '/etc/ejabberd/letsencrypt/' + arguments.domain
+        if not os.path.exists(cert_folder):
+            os.makedirs(cert_folder)
+            shutil.chown(cert_folder, 'ejabberd', 'ejabberd')
+
+        cert_file = cert_folder + '/ejabberd.pem'
+        le_folder = '/etc/letsencrypt/live/' + arguments.domain
+        with open(cert_file, 'w') as outfile:
+            with open(le_folder + '/privkey.pem', 'r') as infile:
+                for line in infile:
+                    if line.strip():
+                        outfile.write(line)
+            with open(le_folder + '/fullchain.pem', 'r') as infile:
+                for line in infile:
+                    if line.strip():
+                        outfile.write(line)
```

Comment:
Me and @SunilMohanAdapa discussed this a bit in today's hack call.
While the upstream issues mentioned are not fixed yet, importantly certbot recently implemented renew hooks, see https://github.com/certbot/certbot/pull/2729. This is all we technically need, so we can move forward to implement something. (Also considering that several modules will need custom solutions, e.g. mumble, and that they need to be restarted afterwards.)

@SunilMohanAdapa pointed out that running the hooks as plinth user, but I don't fully understand. Would be great if more implementation details could be discussed, for anyone who wants to pick this up.